hudi-common/src/main/java/org/apache/hudi/common/table/TableSchemaResolver.java

```diff
@@ -150,6 +150,23 @@ public Schema getTableAvroSchema(String timestamp) throws Exception {
         .orElseThrow(schemaNotFoundError());
   }
 
+  /**
+   * Fetches tables schema in Avro format as of the given instant
+   *
+   * @param timestamp as of which table's schema will be fetched
+   */
+  public Schema getTableAvroSchema(String timestamp, boolean includeMetadataFields) throws Exception {
```

Comment:
I think there is some code reuse we can do with the method above

Reply:
done